### PR TITLE
Update Teams implementation in Developer Guide

### PR DESCRIPTION
**Pull request by mohamedsaf1**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -131,6 +131,22 @@ How the parsing works:
 * When called upon to parse a user command, the `AddressBookParser` class creates an `XYZCommandParser` (`XYZ` is a placeholder for the specific command name e.g., `AddCommandParser`) which uses the other classes shown above to parse the user command and create a `XYZCommand` object (e.g., `AddCommand`) which the `AddressBookParser` returns back as a `Command` object.
 * All `XYZCommandParser` classes (e.g., `AddCommandParser`, `DeleteCommandParser`, ...) inherit from the `Parser` interface so that they can be treated similarly where possible e.g, during testing.
 
+#### Tasks
+
+Commands for tasks (e.g., `AddTaskCommand`, `MarkTaskCommand`, ...) follow a similar sequence of interactions within
+the `Logic` component as the other commands described above.
+
+However, during execution, tasks commands may update the context of the current tasks of a team. To understand tasks better,
+Tasks are analogous to folders in a basic file system, which can contain other folders (`Team`), or other
```
Comment (autumn-sonata) [inline docs/DeveloperGuide.md:140]: This description is for `team`, since `teams` can contain other `teams`, `person` or `task`.

**Review by autumn-sonata — CHANGES_REQUESTED** [review]
Do remember to check through for refactoring errors. Intellij might not be so smart! 😉

**Review by jasonchristopher21 — APPROVED** [review]
LGTM! Perhaps the sequence diagram can be implemented in a future iteration, as this is the DG draft. 👍

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -131,6 +131,23 @@ How the parsing works:
...
+
+However, during execution, tasks commands may update the context of the current tasks of a team. To understand tasks better,
+it should be understood that Tasks are an attribute of a Teams object, which is used to indicate if a tasks is done (also known as marked),
+or not (also known as unmarked).
+
+
+The Sequence Diagram below shows the interaction between `Logic` and `Model` components when `execute("at ..")` is called.
+
+[insert Sequence Diagram]
+
```
Comment (autumn-sonata) [inline docs/DeveloperGuide.md:147]: Do remember to add the sequence diagram for the `execute("at ..")` command.

**Review by autumn-sonata — APPROVED** [review]
LGTM other than the fact that the sequence diagram has not been added. Please add the sequence diagram in the final version of the developer guide.

**Pull request merged**